### ISSUE TITLE
В комментариях админов сделать распознавание ссылок

### ISSUE THREAD

**Issue by mj4444ru** [freedomsex/freedomsex#154]
В комментариях админов при наличии ссылки/ссылок типа http://freedomsex.net/userinfo/profile/12560689/
заменять при сохранении на формат типа
!FDS/userinfo/profile/12560689/
а при отображении !FDS менять на текущий домен и отображать в виде ссылки
естественно это относиться ко всем ссылкам http://freedomsex.net/* и других freedomsex доменов тоже...

Можно даже текст вместо ссылки писать в зависимости от шаблона
Профиль #123456789
Анкета #123456789
Администратор #123456789

**Comment by tebaly** [comment]
Да.

**Comment by tebaly** [comment]
Публикуем исходники сервисов и тестов для них в ветке `dev` перед деплоем. В ветке `news` готовится новость о старте публикации и обзора действующего кода. Переходим к соглашению, по которому ветке `master` соответствует состояние кода на сервере.

https://github.com/tebaly/freedomsex/blob/news/doc/news/20160704-Обзор-кода.md

**Comment by tebaly** [comment]
Parser - находит ссылки и делает их _активными_
Shorter - укорачивает, преобразует ссылки к виду #1234 @4321

**Comment by mj4444ru** [comment]
Так уже написали же? Зачем тема тут нужна?

**Comment by tebaly** [comment]
нет. я намеренно опубликовал код с ошибками, который в принципе работать не может и не работает - ни одного замечания :sweat:

**Comment by mj4444ru** [comment]
было наивно ожидать чего то другого... вы его написали как пример... пример к теме которая на практике никого не интересует, так как нет никакой информации об окружении... логично что никто смотреть не будет...

**Comment by tebaly** [comment]
Какое окружение? Вы программист или тролль 80 уровня?) Абстракции - они ни чего не знают об окружении. ORM - ничего не знает о источнике данных и так далее. Здесь парсер ссылок.. какое парсеру нужно окружение..

Он на входе получает текст, на выходе дает тот же текст со ссылками. Вы ему хоть "войну и мир" пихайте, он отработает и вернет что от него ожидают. А ошибки в JS, которым уже больше года, показывают, на сколько "знающие" персоны заинтересованы в их исправлении :kissing_heart:

**Comment by mj4444ru** [comment]
Не, вы как всегда не поняли о чём я... Ваш пример был бы интересен, если бы кто то хотел написать что то ещё... Но писать что то для вашей системы, это всё равно что писать модуль к фреймворку, о котором ничего не известно, ни документации, ни исходников... Так как модуль не вызывает интереса, никто его и не смотрел... Ошибки в JS я вам нашёл уже (и то лишь по тому что она вызвала ошибку в моём коде)...

**Comment by tebaly** [comment]
> Но писать что то для вашей системы, это всё равно что писать модуль к фреймворку 

Вообще не так)) фреймворк подразумевает переиспользование и распространение. Код для сайта фактически узкоспециализирован изначально, лишь малая его часть может быть использована кем-то ещё в тематически ином проекте. Вы мыслите в понятиях Yii - здесь же вы определяете интерфейс и реализуете его и только. Я его всандалю в проект как вы его не напишите :)

**Comment by mj4444ru** [comment]
Я не вижу задач которые можно было бы сделать в таком варианте... Если вы хотите чьей то помощи, вы должны взаимодействовать, а не ожидать каких то непонятных кусочков бессмысленного кода...
Что бы начать совместную разработку, вам надо открыть большую часть исходного кода, при этом  вы можете заменить какие то фрагменты кода системы безопасности (формулы, что то ещё) на пустышки и тогда люди к вам потянутся... Да и вообще вы не открыты на диалог, вы упрямы и всё сделаете по своему...

**Comment by mj4444ru** [comment]
В конце концов вы даже можете сделать девелоперскую версию сайта на своих серверах, в которую можно будет грузить всем свои изменения через гитхаб и обсуждать с админами нововведения или даже с активными постоянными неравнодушными пользователями...

**Comment by tebaly** [comment]
Нет, вы не правы :stuck_out_tongue:

**Comment by tebaly** [comment]
Вы говорили что в симфони не шарите, а в JS можете. Все исходники JS опубликованы. 
Давайте еще раз. Все исходники javascript опубликованы. Все. Опубликованы.

**Comment by mj4444ru** [comment]
Уже неплохо... Так чего вы ожидаете от этих исходников?
1) Люди будут помогать исправлять ошибки...
2) Предлагать свои варианты дизайна?
3) Укажите свои варианты...

Как вы себе представляете тестирование каких то правок? Учитывая что данные на страницах динамические...

**Comment by mj4444ru** [comment]
Мало того что вы SID 3 раза в сессию суёте, так ещё 4 раз в закодированном виде... Это что бы жизнь малом не казалась?

{"typ":"JWT","alg":"HS256"}{"iss":"freedomsex","exp":1487618864,"sid":"baba3292e89430c8f46d9c92b871405b","uid":"22274890","auth":"22274890","ip":"98.89.37.47","roles":["ROLE_USER"]}
&h6FPY=B0

Мне лень проверять, но смею предположить что можно получить с левого сайта токен авторизации и запросить по нему список сообщений, передав токен в заголовке Authorization и т.д., то есть убив всю ту безопасность, к которой так долго шли создатели браузеров...

**Comment by tebaly** [comment]
JWT - новая технология. Запросить токен можно хоть с утюга, но при условии, что вы с него авторизуетесь на сайте. Получает токен только авторизованный пользователь. И дальше, имея токен - да, иметь информацию можно откуда угодно. Токен подписан.

**Comment by tebaly** [comment]
Интереснее другое. Я по незнанке, доверившись babe(l) употребил в коде _Object.assign_ - и пол дня подрывал пуканы всех, кто не пользует евергрин браузеры :grimacing:

**Comment by mj4444ru** [comment]
Почитал про JWT... Ну в принципе ничего так стандарт, наверно кому то полезный... Но вам то он зачем???

> Object.assign
Именно по этому без крайней необходимости не юзаю голый JS, стараюсь доверять фреймворку... Кажется во всех JS фреймворках есть аналог этой функции...

А так я уже лет 5 на JS особо не пишу... Сейчас в магенте капаюсь, что откуда и как приходит... В knockoutjs разобрался, теперь в php коде капаюсь... Это просто пи...

**Comment by tebaly** [comment]
@mj4444ru - всё не то. Vue вперед! Наикрутейшая вещь. 
Это я вам после дегустации Эмбера, Бэкбона и Реакта говорю :heart_eyes:

**Comment by tebaly** [comment]
`Но вам то он зачем???` - крайне нужно и необходимо. Новогодний подарок конечно задерживается, но его ни кто не отменял)))

**Comment by mj4444ru** [comment]
Если это всё ради фотографий, то это вы зря... Я бы не стал позволять сторонним сайтам получать доступ к данным пользователя... Это можно использовать в плохом виде... Я даю пользователю ссылку, на сайт FDS, только с более красивым дизайном, пользователь там авторизуется и я получаю всю его переписку... То есть сайт злоумышленника будет тупо статикой, а все запросы будут идти напрямую от клиента... Хотя не исключаю что это можно обойти через сторонний сервер-шлюз, но это уже сложнее...

Если вам эта штука нужна для чего то (фото сервер, кривой телефонный клиент), то использовали ба только там...

**Comment by tebaly** [comment]
Чем это отличается от куки? 
Вы через FDS можете и логин с паролем выхватить.

**Comment by tebaly** [comment]
Я могу неверно использовать технологию. Но говорить о том, что технология не нужна - глупо. Я уже как несколько лет назад попросил Пользователей сайта не использовать в общении сильно компрометирующей или сверх ценной информации. Я обеспечиваю безопасность в проекте на максимально возможном для меня уровне - но это не уровень совершенства. Информация у нас должна обладать свойствами
* отсутствия ценности - всё может быть удалено в любой момент
* не конфиденциальна - анонимна или бессмысленна в случае перехвата

Рядовой пользователь доступа не имеет. А хакеры, они как воры или грабители - нет никаких гарантий, что вас не ограбят на улице или не вычистят вашу квартиру пока вы на работе. Базы данных крупнейших корпораций в интернете регулярно появляются в сети. Наш сайт крупинка. И если у них беда с безопасностью. Имею ли я право давать гарантии?..

**Comment by mj4444ru** [comment]
Да дело не в обеспечении безопасности, а в целесообразности... Просто пытаюсь понять зачем оно вам тут нужно... Что оно вам даёт... Как мне кажется. достаточно хранить в куках токен авторизации пользователя... Всю остальную инфу лучше передавать в теле страницы, нет необходимости захламлять запросы ненужными куками...

**Comment by tebaly** [comment]
А с сервером фоток как предлагаете разговаривать?? Он куку не примет!

**Comment by tebaly** [comment]
Всё остальное - обратная совместимось, для тех у кого старые куки. Удалю конечно дубли. Но пока они не мешают
